src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -799,10 +799,14 @@ def _update_causal_mask(
         # order to dispatch on Flash Attention 2. This feature is not compatible with static cache, as SDPA will fail
         # to infer the attention mask.
         past_seen_tokens = past_key_values.get_seq_length() if past_key_values is not None else 0
-        using_static_cache = isinstance(past_key_values, StaticCache)
 
         # When output attentions is True, sdpa implementation's forward method calls the eager implementation's forward
-        if self.config._attn_implementation == "sdpa" and not using_static_cache and not output_attentions:
+        if (
+            self.config._attn_implementation == "sdpa"
+            and past_key_values is not None
+            and not past_key_values.is_static
```

Comment:
Aligned with my previous comment where we need a general `past_key_value.skip_sdpa_correction()` for example

Reply:
Hmm, maybe we can make a better check to not use `isinstance()` on cache objects. We are having more varieties of cache now and it is mostly either dynamic or static type of some caching method, so it would be nice to not rely on `isinstance()` checks in general

We can remove modeling changes in this PR and use the new attributes in specific cases (linked comment from PR description). What I need from this PR currently is the `get_max_length` that doesn;t return `None` anymore

Reply:
Yeah Also want to avoid .is_static and isinstance etc. Not sure how to abstract best tho! 

Okay, let's remove then for now

Reply:
okey, done and rebased main